### PR TITLE
#4003: implemented functional t5 model

### PR DESCRIPTION
**Pull request by arakhmati**

### CODE REVIEW

**Review by eyonland — APPROVED** [review]

File: models/utility_functions.py [hunk 1]
```diff
@@ -50,6 +50,8 @@ def float_to_bits(x):
 
 
 def torch_random(shape, low, high, dtype):
+    if dtype == torch.int64:
```
Comment (tt-aho) [inline models/utility_functions.py:53]: should we use is_floating_point here to check?

**Review by tt-aho — APPROVED** [review]
lgtm

**Review by farbabi — APPROVED** [review]

File: models/utility_functions.py (hunk 1 above)
Comment (arakhmati) [reply models/utility_functions.py:53]: Good idea! I will do that

Comment (arakhmati) [reply models/utility_functions.py:53]: Unfortunately, it only works on tensors. Not on dtype objects

Comment (tt-aho) [reply models/utility_functions.py:53]: It should work for dtypes, it's a property
image

**Pull request merged**